### PR TITLE
Update index.d.ts

### PR DESCRIPTION
**Pull request by Agapanthus**
Added defaults and getTokenTypeAt

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [ ] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://codemirror.net/doc/manual.html
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/codemirror/index.d.ts [hunk 1]
```diff
@@ -315,8 +321,13 @@ declare namespace CodeMirror {
...
         getTokenAt(pos: CodeMirror.Position, precise?: boolean): Token;
 
+        /** This is a (much) cheaper version of getTokenAt useful for when you just need the type of the token at a given position, 
+        and no other information. Will return null for unstyled tokens, and a string, 
+        potentially containing multiple space-separated style names, otherwise. */
+        getTokenTypeAt(pos: CodeMirror.Position): string;
+
         /** This is similar to getTokenAt, but collects all tokens for a given line into an array. */
-        getLineTokens(line: number, precise?: boolean): Token[];
+        getLineTokens(line: number, precise?: boolean): Token[];
```
Comment (mihailik) [inline types/codemirror/index.d.ts]: Let's cleanup this please.

**Review by mihailik — CHANGES_REQUESTED** [review]
Given this is a cleanup, let's make sure stray whitespace isn't added. Otherwise 100% good to go.

Thanks @Agapanthus great work!

**Comment by Agapanthus** [comment]
Ok, I removed that whitespace.

**Comment by mihailik** [comment]
Thanks a lot LGTM :+1:

**Comment by mihailik** [comment]
Sorry, just realised - the Travis build fails because there's another invisible whitespace at the beginning of the file.

You can see it says  "All checks have failed" at the bottom.

<a href="https://travis-ci.org/DefinitelyTyped/DefinitelyTyped/builds/386229772#L1748">Travis build issue:</a>

```
Error: File '/home/travis/build/DefinitelyTyped/DefinitelyTyped/types/codemirror/index.d.ts' has a BOM. Try using:
```

What editor are you using — I know Visual Studio can include those BOM markers, but you can tell it to skip if you do "Save As" and pick the encoding something like "UTF8 without BOM".

Sorry I didn't spot it the first time around — would have been easier for you to fix both issues. Although this BOM thing is tiny, it can mess things up in a bad way.

**Comment by Agapanthus** [comment]
Im sorry - i thought removing a single whitespace character using the integrated text editor would be ok... Now I removed the BOM.

**Comment by mihailik** [comment]
Check passed, all steam ahead, thanks a lot! :+1: 

LGTM

**Pull request merged**